src/CSharp/CodeCracker/Refactoring/IntroduceFieldFromConstructorAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace CodeCracker.Refactoring
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class IntroduceFieldFromConstructorAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Consider introduce field for constructor parameters.";
+        internal const string MessageFormat = "Introduce a field for parameter: {0}";
+        internal const string Category = SupportedCategories.Refactoring;
+        const string Description = "Consider introduce field for constructor parameters.";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId.IntroduceFieldFromConstructor.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Hidden,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLink: HelpLink.ForDiagnostic(DiagnosticId.TaskNameAsync));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeConstructor, SyntaxKind.ConstructorDeclaration);
+
+        private static void AnalyzeConstructor(SyntaxNodeAnalysisContext context)
+        {
+            var constructorMethod = (ConstructorDeclarationSyntax)context.Node;
+            var parameters = constructorMethod.ParameterList.Parameters;
+
+            if (constructorMethod.Body == null) return;
+            if (constructorMethod?.Body.Statements.Count == 0 && parameters.Count > 0)
+            {
+                foreach (var par in parameters)
+                {
+                    var errorMessage = par.Identifier.Text;
+                    var diag = Diagnostic.Create(Rule, constructorMethod.GetLocation(), errorMessage);
+                    context.ReportDiagnostic(diag);
+                }
+                return;
+            }
```

Comment:
Very smart to take a short route and just report on every parameter if there are no statements!